pennylane/labs/resource_estimation/ops/op_math/controlled_ops.py

```diff
@@ -291,37 +630,97 @@ def controlled_resource_decomp(
     r"""Resource class for the CNOT gate.
 
+    Args:
+        wires (Sequence[int]): the wires the operation acts on
+
     Resources:
-        There is no further decomposition provided for this gate.
+        The CNOT gate is treated as a fundamental gate and thus it cannot be decomposed
+        further. Requesting the resources of this gate raises a :code:`ResourcesNotDefined` error.
+
+    .. seealso:: :class:`~.CNOT`
 
     """
 
     @staticmethod
     def _resource_decomp(**kwargs) -> Dict[re.CompressedResourceOp, int]:
+        r"""Returns a dictionary representing the resources of the operator. The
+        keys are the operators and the associated values are the counts.
+
+        Resources:
+            The CNOT gate is treated as a fundamental gate and thus it cannot be decomposed
+            further. Requesting the resources of this gate raises a :code:`ResourcesNotDefined` error.
+
+        Raises:
+            ResourcesNotDefined: This gate is fundamental, no further decomposition defined.
+        """
         raise re.ResourcesNotDefined
 
     @property
     def resource_params(self) -> dict:
+        r"""Returns a dictionary containing the minimal information needed to compute the resources.
+
+        Returns:
+            dict: Empty dictionary. The resources of this operation don't depend on any additional parameters.
+        """
         return {}
 
     @classmethod
     def resource_rep(cls) -> re.CompressedResourceOp:
+        r"""Returns a compressed representation containing only the parameters of
+        the Operator that are needed to compute a resource estimation."""
         return re.CompressedResourceOp(cls, {})
 
     @classmethod
     def adjoint_resource_decomp(cls) -> Dict[re.CompressedResourceOp, int]:
+        r"""Returns a dictionary representing the resources for the adjoint of the operator.
+
+        Resources:
+            This operation is self-adjoint, so the resources of the adjoint operation results
+            in the original operation.
+
+        Returns:
+            Dict[CompressedResourceOp, int]: The keys are the operators and the associated
+                values are the counts.
+        """
         return {cls.resource_rep(): 1}
 
     @classmethod
     def controlled_resource_decomp(
         cls, num_ctrl_wires, num_ctrl_values, num_work_wires
     ) -> Dict[re.CompressedResourceOp, int]:
+        r"""Returns a dictionary representing the resources for a controlled version of the operator.
+
+        Args:
+            num_ctrl_wires (int): the number of qubits the operation is controlled on
+            num_ctrl_values (int): the number of control qubits, that are controlled when in the :math:`|0\rangle` state
+            num_work_wires (int): the number of additional qubits that can be used for decomposition
+
+        Resources:
+            The resources are expressed as one general :class:`~.ResourceMultiControlledX` gate.
```

Comment:
Or a toffoli gate. Maybe should mention this too?

Reply:
No need to mention the Toffoli, the multi controlled gate will simplify to that case as well.